src/base/utils/randomlayer_linux.cpp

```diff
@@ -56,22 +78,36 @@ namespace
             return std::numeric_limits<result_type>::max();
         }
 
-        result_type operator()()
+        result_type operator()() const
         {
             const int RETRY_MAX = 3;
 
             for (int i = 0; i < RETRY_MAX; ++i)
             {
                 result_type buf = 0;
-                const ssize_t result = ::getrandom(&buf, sizeof(buf), 0);
-                if (result == sizeof(buf))  // success
-                    return buf;
 
-                if (result < 0)
-                    qFatal("getrandom() error. Reason: %s. Error code: %d.", std::strerror(errno), errno);
+                if (!m_randDev)
```

Comment:
IMO, it is better to check `m_randDev` outside of loop.

Reply:
Ok but I don't understand how would you change the code.

Reply:
```c++
result_type operator()() const
{
    if (!m_randDev)
	{
        const int RETRY_MAX = 3;

	    for (int i = 0; i < RETRY_MAX; ++i)
	    {
		    result_type buf = 0;

			const ssize_t result = ::getrandom(&buf, sizeof(buf), 0);
			if (result == sizeof(buf))  // success
				return buf;

			if (result < 0)
				qFatal("getrandom() error. Reason: \"%s\". Error code: %d.", std::strerror(errno), errno);
		}

        qFatal("RandomLayer::operator() failed. Reason: too many retries.");
    }
	else
	{
		if (fread(&buf, sizeof(buf), 1, m_randDev) == 1)
			return buf;

		qFatal("Read /dev/urandom error. Reason: \"%s\". Error code: %d.", std::strerror(errno), errno);
	}
}
```

Reply:
You can even extract different implementations into separate functions.

Reply:
Done.